lua/packer.lua

```diff
@@ -167,6 +168,108 @@ packer.use_rocks = function(rock)
   end
 end
 
+-- Merges new into old; returns merged spec.
+local function merge_specs(old, new)
+  local name = new.short_name
+
+  -- disable: Warn & ignore on disabling a previously enabled plugin
+  if old.disable then
+    if not new.disable then log.warn('Trying to disable previously non-disabled mod "' .. name .. '", ignoring') end
+  else
+    old.disable = new.disable
+  end
+
+  -- branch, tag, commit, lock: Must all be the same, else warning
+  for _, key in ipairs{'branch', 'commit', 'lock', 'tag'} do
+    if new[key] ~= old[key] then
+      log.warn('Plugin "' .. name .. '" wants to overwrite "' .. key .. '" setting. Keeping original value')
```

Comment:
Is it worth printing the two values in this warning, e.g. to make determining which spec is "new" easier for users?

Reply:
Yeah, it'd definitely be useful. Last night I finally understood what the `plugin_spec.line` and IIUC we could use it to highlight the offending lines here.